test/controllers/ATSPAYEDataControllerTest.scala

```diff
@@ -16,57 +16,113 @@
+  val expectedResponseCYPlus1 = PayeAtsMiddleTier(cyPlus1, testNino, None, None, None, None, None)
 
-    val expectedResponse = PayeAtsMiddleTier(2018, testNino, None, None, None, None, None)
+  "getAtsData" should {
 
     "return success response with ATS data" in new TestController {
-      when(npsService.getPayeATSData(eqTo(testNino), eqTo(2018))(any[HeaderCarrier]))
-        .thenReturn(Right(expectedResponse))
-      val result = getATSData(testNino, 2018)(request)
+      when(npsService.getPayeATSData(eqTo(testNino), eqTo(cy))(any[HeaderCarrier]))
+        .thenReturn(Right(expectedResponseCY))
+      val result = getATSData(testNino, cy)(request)
 
       status(result) shouldBe 200
-      contentAsJson(result) shouldBe Json.toJson(expectedResponse)
+      contentAsJson(result) shouldBe Json.toJson(expectedResponseCY)
     }
 
     "return a failed future" in new TestController {
-      when(npsService.getPayeATSData(eqTo(testNino), eqTo(2018))(any[HeaderCarrier]))
-        .thenReturn(Left(HttpResponse(responseStatus = BAD_REQUEST, responseJson = Some(Json.toJson(BAD_REQUEST)))))
-      val result = getATSData(testNino, 2018)(request)
+      when(npsService.getPayeATSData(eqTo(testNino), eqTo(cy))(any[HeaderCarrier]))
+        .thenReturn(Left(HttpResponse(BAD_REQUEST, BAD_REQUEST.toString)))
+      val result = getATSData(testNino, cy)(request)
 
       status(result) shouldBe 400
       contentAsJson(result) shouldBe Json.toJson(BAD_REQUEST)
     }
   }
+
+  "getATSDataMultipleYears" should {
+
+    "return a list of data for the given tax years" in new TestController {
+      when(npsService.getPayeATSData(eqTo(testNino), eqTo(cy))(any[HeaderCarrier])) thenReturn Future.successful(
+        Right(expectedResponseCY))
+      when(npsService.getPayeATSData(eqTo(testNino), eqTo(cyPlus1))(any[HeaderCarrier])) thenReturn Future.successful(
+        Right(expectedResponseCYPlus1))
+
+      val result = getATSDataMultipleYears(testNino, cy, cyPlus1)(request)
+
+      status(result) shouldBe OK
+      contentAsJson(result) shouldBe Json.toJson(Seq(expectedResponseCY, expectedResponseCYPlus1))
+    }
+
+    "return NOT_FOUND" when {
+
+      "the service returns no data" in new TestController {
+
+        when(npsService.getPayeATSData(eqTo(testNino), eqTo(cy))(any[HeaderCarrier])) thenReturn Future.successful(
+          Left(HttpResponse(BAD_REQUEST, "Not found")))
+        when(npsService.getPayeATSData(eqTo(testNino), eqTo(cyPlus1))(any[HeaderCarrier])) thenReturn Future.successful(
+          Left(HttpResponse(BAD_REQUEST, "Not found")))
+
+        val result = getATSDataMultipleYears(testNino, cy, cyPlus1)(request)
+
+        status(result) shouldBe NOT_FOUND
+        bodyOf(result).futureValue shouldBe s"No data found for $testNino"
+      }
+    }
+
+    "return INTERNAL_SERVER_ERROR" when {
+
+      "an exception occurs when retrieving data" in new TestController {
+
+        val exMessage = "Oh dear"
```

Comment:
lol
I dont mind but I think some people might not like this so
can we please provide an appropriate exception message here?